sdk/search/search-documents/scripts/generateTestVarNames.pl

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env -S perl -i
+
+use Algorithm::Combinatorics qw(variations_with_repetition);
```

Comment:
what is this for, and why not use JavaScript? 😈

Reply:
It's a small script to save me some time while hand-editing the `typeDefinitions.ts` test files. It just renames every const variable declaration with sequential strings. There's more that I intended to automate in that space, but I think I either lost some of that progress in a Codespace rebuild or otherwise shelved that for other priorities. 

As for why it's Perl, I find that text file manipulation is a bit more idiomatic. `perl -i` modifies files in-place, so it's common to find Perl one-liners in the form `perl -i -pe 's/foo/bar/msg' foo.txt` (replaces all occurrences of `foo` with `bar`).

Reply:
FWIW we used to have a small `.js` build script for EventGrid that did some massaging of types. I think in general it'd be good to stick to JS/TS when possible so you can hand it off to other maintainers more easily.